src/main/java/com/google/sps/data/Event.java

```diff
@@ -0,0 +1,189 @@
+ 
+  // inputs
+  private String name;
+  private String address;
+  private String date;
+  // format: HHMM
+  private int startTime;
+  private int endTime;
+
+  //in mins
+  private int travelTime;
+
+  // fields to be saved for parsing
+  private String strStartTime;
+  private String strEndTime;
+ 
+  // class constants
+  private static final int HALFHOUR = 30;
+  private static final int HOUR = 60;
+  private static final int MINUTES_IN_A_DAY = 1440;
+  private static final int MIN_POSSIBLE_TIME = 0;
+
+  /**
+   * Constructor that takes in time spent at location
+   * 
+   * @param name name of the location (NOT address)
+   * @param address exact address of the POI
+   * @param date date from user input travel day. Cannot be null.
+   * @param startTime start of activity (HHMM format)
+   * @param travelTime time spent traveling to next location (minutes)
+   * @param timeAtLocation time spent at POI (minutes)
+   */
+  public Event(String name, String address, String date, int startTime, 
+              int travelTime, int timeAtLocation) {
+    this.name = name;
+    this.address = address;
+    this.date = date;
+    this.startTime = startTime;
+    this.travelTime = travelTime;
+    this.endTime = calculateEndTime(timeAtLocation);
+
+    createCalendarTimes();
+  }
+ 
+  /**
+   * Constructor that assume one hour default spent at location
+   * 
+   * @param name name of the location (NOT address)
+   * @param address exact address of the POI
+   * @param date date from user input travel day. Cannot be null.
+   * @param startTime start of activity (HHMM format)
+   * @param travelTime time spent traveling to next location (minutes)
+   */
+  public Event(String name, String address, String date, int startTime, int travelTime) {
+    this.name = name;
+    this.address = address;
+    this.date = date;
+    this.startTime = startTime;
+    this.travelTime = travelTime;
+    this.endTime = calculateEndTime(HOUR);
+
+    createCalendarTimes();
+  }
+
+  // function that sets start and end Time with correct string format
+  private void createCalendarTimes() {
+    this.strStartTime = createStrTime(this.date, this.startTime);
+    this.strEndTime = createStrTime(this.date, this.endTime);
+  }
+
+  /**
+   * function that builds string w correct format for calendar-script.js
+   * 
+   * @param date date from user input travel day. Cannot be null.
+   * @param time time to to start or end event. Must be within bounds
+   */
+  private static String createStrTime(String date, int time) {
+    if (date == null) {
+      throw new NullPointerException("Date cannot be null");
+    }
+
+    if (time < MIN_POSSIBLE_TIME) {
+      throw new IllegalArgumentException("Time cannot be less than 0");
+    }
+
+    if (time >= MINUTES_IN_A_DAY) {
+      throw new IllegalArgumentException("Time cannot be more than a 24hrs");
+    }
+
+    String output = "";
+    output += date + "T" + Integer.toString(time).substring(0, 2) + ":" + 
```

Comment:
Consider using a date parsing library like SimpleDateFormat [link: https://docs.oracle.com/javase/8/docs/api/java/text/SimpleDateFormat.html#parse-java.lang.String-java.text.ParsePosition-] to outsource this parsing logic.

Using a date library will also take care of edge cases like daylight savings time, leap years etc.

If your work is blocking on this CL I'd be ok with submitting it as-is, but I think I'd prefer to switch to a date library in a follow-up CL.

Reply:
Would love to talk about this in over the phone tomorrow when you have the chance!